app/models/mixins/relationship_mixin.rb

```diff
@@ -682,4 +755,12 @@ def detect_ancestor(*args, &block)
   def puts_relationship_tree
     Relationship.puts_arranged_resources(subtree_arranged)
   end
+
+  private
+
+  # this allows us to call similarly named ancestry or mixin methods
+  def call_ancestry_method(method_name, *args)
+    bound_method = ((@@ancestry_method ||= {})[method_name] ||= ::Ancestry::InstanceMethods.instance_method(method_name)).bind(self)
+    args.empty? ? bound_method.call : bound_method.call(*args)
+  end
```

Comment:
NOTE: C++ and Java allow you to do this much more easily, calling a specific super implementation.

this looks up the method in the ancestry code and calls the specific method on self.
This kinda lets us have 2 parent implementations based upon the relationship_type.